docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+  table=4 (ls_out_acl_eval    ), priority=27599, match=(reg8[30..31] == 1 && reg0[8] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a6786643370959569281)) && tcp && tcp.dst==7564)), action=(log(name="ANP:cluster-control:Ingress:1", severity=alert, verdict=allow, meter="acl-logging__4117970c-c138-45b5-a6de-8e5e54efcb04"); reg8[16] = 1; next;)
+  table=4 (ls_out_acl_eval    ), priority=27598, match=(reg8[30..31] == 1 && reg0[7] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a13730899355151937870)))), action=(log(name="ANP:cluster-control:Ingress:2", severity=alert, verdict=allow, meter="acl-logging__00400c05-7225-4d64-9428-8b2a04c6c3da"); reg8[16] = 1; reg0[1] = 1; next;)
+  table=4 (ls_out_acl_eval    ), priority=27598, match=(reg8[30..31] == 1 && reg0[8] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a13730899355151937870)))), action=(log(name="ANP:cluster-control:Ingress:2", severity=alert, verdict=allow, meter="acl-logging__00400c05-7225-4d64-9428-8b2a04c6c3da"); reg8[16] = 1; next;)
+  table=4 (ls_out_acl_eval    ), priority=27597, match=(reg8[30..31] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a764182844364804195)))), action=(log(name="ANP:cluster-control:Ingress:3", severity=warning, verdict=pass, meter="acl-logging__7f5ea19b-d465-4de4-8369-bdf66fdcd7d9"); next;)
+  table=4 (ls_out_acl_eval    ), priority=27596, match=(reg8[30..31] == 1 && reg0[10] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a13814616246365836720)))), action=(log(name="ANP:cluster-control:Ingress:4", severity=alert, verdict=drop, meter="acl-logging__3e82df77-ad40-4dce-bd96-5b9e7e0b3603"); reg8[17] = 1; ct_commit { ct_mark.blocked = 1; }; next;)
+  table=4 (ls_out_acl_eval    ), priority=27596, match=(reg8[30..31] == 1 && reg0[9] == 1 && (outport == @a14645450421485494999 && ((ip4.src == $a13814616246365836720)))), action=(log(name="ANP:cluster-control:Ingress:4", severity=alert, verdict=drop, meter="acl-logging__3e82df77-ad40-4dce-bd96-5b9e7e0b3603"); reg8[17] = 1; next;)
+  table=4 (ls_out_acl_eval    ), priority=2750 , match=(reg8[30..31] == 3 && reg0[10] == 1 && (outport == @a9550609891683691927 && ((ip4.src == $a168374317940583916)))), action=(reg8[17] = 1; ct_commit { ct_mark.blocked = 1; }; next;)
+  table=4 (ls_out_acl_eval    ), priority=2750 , match=(reg8[30..31] == 3 && reg0[9] == 1 && (outport == @a9550609891683691927 && ((ip4.src == $a168374317940583916)))), action=(reg8[17] = 1; next;)
 ```
+
+## Multi Tenant Isolation
+
+In order to isolate your tenants in the cluster, unlike
+network policies you don't need to create 1 policy per namespace.
+You need to only create 1 policy per tenant. Sample ANPs for
+multi-tenant isolation where we have 4 tenants `blue`, `green`,
+`yellow` and `red` are:
```

Comment:
With these policies, all traffic within a tenant is forcibly Allowed; namespace admins are not permitted to have any within-tenant security. IMO, the more expected behavior for "multi-tenant isolation" would be a policy for `red` to `Deny` traffic not from `red`, a policy for `blue` to `Deny` traffic not from `blue`, etc, with no `Allow` rules.

(Admins almost never want to `Allow` arbitrary traffic. `Allow` in ANP does not mean "I want this allowed by default", it means "I want this to be allowed even if the user wants to block it". If you aren't explicitly trying to prevent users from blocking the traffic, then you mean `Pass`, not `Allow`.)

Reply:
oh that's a great point!
so the expression of deny from everything else at prio 43 was meant to act like that deny rest of the traffic. So I have 5 ANPs at prio 40 that allow and then the ANP at prio 43 denies everything else.
So essentially its the same as "deny not from yourself" -> But I am also going to include your way of doing things.. that sounds like another second way to express it. So the way I had thought of this was allow whatever you want to allow at a higher prio and then deny everything else.

Reply:
> So essentially its the same as "deny not from yourself"

No, it's not. It's "deny not from yourself _and you aren't allowed to deny anything else_".

The entire point of NP/ANP is that admins like to deny traffic. If pod A is not supposed to talk to pod B, then many admins want to be able to set policies so that pod A _can't_ talk to pod B.

But if you create an ANP that says "`Allow` from blue to blue", then Game Over. It is no longer possible to create an NP that says that pod A and pod B can't talk to each other, because the ANP would override it. So if a hacker can break into the most insecure pod in the tenant, then they can use that to reach every other pod in the tenant.

No cluster admin wants to create a policy that prevents project admins from being able to follow good security practices within their own namespaces.

Reply:
sure if they want finer grained controls within the tenant they can, its all about the design ultimately, the requirements I portrayed here is "allow all pods in blue to be able to talk to all pods in blue" within them I don't want to restrict anything, so I can use allow.

If users want to put "within-tenant-further-control" then your scenario works and mine won't. I will document that.

Reply:
I have clarified this point too, thanks Dan for raising this!